src/adapters/amqp.js

```diff
@@ -434,22 +468,10 @@ class AmqpAdapter extends BaseAdapter {
 	 * @param {Object} msg
 	 */
 	async moveToDeadLetter(chan, msg) {
```

Comment:
All the adapters have the `HEADER_ORIGINAL_CHANNEL` and `HEADER_ORIGINAL_GROUP` in headers. Removing this logic is something that we would like to avoid in here.

Can you please revert to the original logic?

Reply:
@AndreMaz I am a little confused. I didn't actually remove this. I just moved it up in the method so that the dead letter exchange/queue logic could use it too. Am I missing something?

Reply:
@AndreMaz, the reason I did this is that this was actually not working at high data rates, when I was pumping in 10,000 messages/second, and I was throwing an error in the handler to trigger the movement of the messages into the deadletter queue. The republish would cause a buffer full, and messages would get stuck and the consumer handler would not get triggered. I am not sure what the problem was, but using the native requeueing vs republishing seemed to work. I can revert this, but there will be an issue at high loads.